module-bsp/board/rt1051/bsp/lpm/RT1051LPMCommon.cpp

```diff
@@ -178,27 +178,30 @@ namespace bsp
 
     void RT1051LPMCommon::RegularLDOMode()
     {
-        // Enable regular 2P5 and wait it stable
+        // Enable regular 2P5 and 1P1 LDO
         PMU->REG_2P5_SET = PMU_REG_2P5_ENABLE_LINREG_MASK;
-        // It is recommended to wait for stabilization (documentation Low Power AN12085)
-        while ((PMU->REG_2P5 & PMU_REG_2P5_OK_VDD2P5_MASK) == 0) {}
-        // Turn off weak 2P5
-        PMU->REG_2P5_CLR = PMU_REG_2P5_ENABLE_WEAK_LINREG_MASK;
-        // Enable regular 1P1 and wait for stable
         PMU->REG_1P1_SET = PMU_REG_1P1_ENABLE_LINREG_MASK;
-        // It is recommended to wait for stabilization (documentation Low Power AN12085)
+
+        // Wait for regular LDOs to become stable (documentation Low Power AN12085)
+        while ((PMU->REG_2P5 & PMU_REG_2P5_OK_VDD2P5_MASK) == 0) {}
         while ((PMU->REG_1P1 & PMU_REG_1P1_OK_VDD1P1_MASK) == 0) {}
-        // Turn off weak 1P1
+
+        // Turn off weak 2P5 and 1P1 LDO
+        PMU->REG_2P5_CLR = PMU_REG_2P5_ENABLE_WEAK_LINREG_MASK;
         PMU->REG_1P1_CLR = PMU_REG_1P1_ENABLE_WEAK_LINREG_MASK;
     }
 
     void RT1051LPMCommon::LowPowerLDOMode()
     {
-        // Enable weak 2P5 and turn off regular 2P5
-        PMU->REG_2P5 |= PMU_REG_2P5_ENABLE_WEAK_LINREG_MASK;
-        PMU->REG_2P5 &= ~PMU_REG_2P5_ENABLE_LINREG_MASK;
-        // Enable weak 1P1 and turn off regular 1P1
-        PMU->REG_1P1 |= PMU_REG_1P1_ENABLE_WEAK_LINREG_MASK;
-        PMU->REG_1P1 &= ~PMU_REG_1P1_ENABLE_LINREG_MASK;
+        // Enable weak 2P5 and 1P1 LDO
+        PMU->REG_2P5_SET = PMU_REG_2P5_ENABLE_WEAK_LINREG_MASK;
+        PMU->REG_1P1_SET = PMU_REG_1P1_ENABLE_WEAK_LINREG_MASK;
```

Comment:
as you're using the same mask for CLR and SET registers, maybe it's worth removing `ENABLE` from the mask?

Reply:
Those masks are used by the fsl SDK so we won't to change it (maybe in the future we'll update the fsl SDK) ;)

Reply:
alright :)